File: modules/01-basic-syntax/go/main.go
package main

import (
	"myapp/src/modules"
	"myapp/src/utils"
)

func main() {
	utils.PrintH1("Basic syntax")

  // Compound types
	utils.PrintH2("Compound types")
	modules.Array_asignment_and_access()
	modules.Tuple_assignment_and_access()
	utils.PrintBr()

	// References
	utils.PrintH2("References")
	modules.References()
	utils.PrintBr()

	// Slices
	utils.PrintH2("Slices")
	modules.Array_slices()
	modules.String_vs_str()
	utils.PrintBr()

	// Functions
	utils.PrintH2("Functions")
	modules.Fizzbuzz_to(10)
	modules.Test_methods_from_rectangle()
	modules.Test_overloading()
	utils.PrintBr()
}

File: modules/01-basic-syntax/rust/src/main.rs
mod modules; // this declares a root module called `modules`
mod utils;

fn main() {
    let print_h1 = utils::print_h1;
    let print_h2 = utils::print_h2;
    let print_br = utils::print_br;

    print_h1("Basic syntax");

    // Compound types
    print_h2("Compund types");
    modules::compound_types::array_asignment_and_access();
    modules::compound_types::tuple_assignment_and_access();
    print_br();

    // References
    print_h2("References");
    modules::references::references();
    print_br();

    // Slices
    print_h2("Slices");
    modules::slices::array_slices();
    modules::slices::string_vs_str();
    print_br();

    // Functions
    print_h2("Functions");
    modules::functions::fizzbuzz_to(10);
    modules::functions::test_methods_from_rectangle();
    modules::functions::test_overloading();
    print_br();
}
